### PR TITLE
[Heap]: Update the action-destination to use integrations endpoint 

### PR DESCRIPTION
**Pull request by murphpdx**
I've updated the action-destination to use our internal integrations endpoint rather than the public one. There is additional functionality that we'd like to leverage in the integrations endpoint. This endpoint only takes in strings for properties so I've updated the flatten function to convert them. I've also updated how we take in anonymous_id so that we can better merge users. 

## Testing

I've used the action tester to verify that events are coming into heap as expected.

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
hi @murphpdx just acknowledging this PR. I'll get to is ASAP.

**Comment by joe-ayoub-segment** [comment]
hi @murphpdx would you be available to meet to review this together? Keen to ensure nothing breaks for existing customers. 
Could you schedule a call with me please http://calendly.com/joe_ayoub/integration-workshop ?

File: packages/destination-actions/src/destinations/heap/trackEvent/__tests__/__snapshots__/snapshot.test.ts.snap [hunk 1]
```diff
@@ -3,23 +3,60 @@
 exports[`Testing snapshot for Heap's trackEvent destination action: all fields 1`] = `
 Object {
   "app_id": "xqYHVWXiU0In",
+<<<<<<< HEAD
```
Comment (NolanChan) [inline packages/destination-actions/src/destinations/heap/trackEvent/__tests__/__snapshots__/snapshot.test.ts.snap:6]: Hi @murphpdx ! I think you left some conflicts in here by accident

**Comment by murphpdx** [comment]
> hi @murphpdx would you be available to meet to review this together? Keen to ensure nothing breaks for existing customers. Could you schedule a call with me please http://calendly.com/joe_ayoub/integration-workshop ?

Hi @joe-ayoub-segment I responded to you on slack. None of those times work for me. That's 2am-7am my time. I can do after 9am pacific time.

**Comment by murphpdx** [comment]
We need to do some more verification on this. I will open up a new PR when we're ready.

**Pull request closed without merging**